app/Domain/Integrations/Models/IntegrationModel.php

```diff
@@ -37,9 +37,12 @@ protected static function booted(): void
         self::created(
             static fn ($integrationModel) => IntegrationCreated::dispatch(Uuid::fromString($integrationModel->id))
         );
-        self::deleted(
-            static fn ($integrationModel) => $integrationModel->update(['status' => IntegrationStatus::Deleted])
-        );
+    }
+
+    public function delete(): ?bool
+    {
+        $this->update(['status' => IntegrationStatus::Deleted]);
+        return parent::delete();
```

Comment:
I'm curious, was this necessary to make something work in the ActivityLog library or was the change needed for some other reason? It's not possible to do it via the event anymore?

Reply:
We moved the status code update to the `delete` method to ensure it no longer depends on the `deleted` model event being fired. When doing a `delete` the status should always be updated, it is not a side effect that should be handled when the `deleted` events triggers.